packages/components/src/internal/components/assay/actions.ts

```diff
@@ -41,12 +42,24 @@ export const RUN_PROPERTIES_REQUIRED_COLUMNS = SCHEMAS.CBMB.concat(
     'DataOutputs',
     'DataOutputs/DataFileUrl',
     'Batch',
+    'Folder',
     // Below Columns are required for us to render the WorkflowTask in EditableDetails components
     'WorkflowTask',
     'WorkflowTask/Run',
     'Protocol/RowId'
 ).toList();
 
+/**
+ * Only support option to re-import run if user has insert permissions in current container
+ * and the current container is where the source run is located. This prevents runs from being
+ * re-imported "up" or "down" the folder structure.
+ */
+export function allowReimportAssayRun(user: User, runContainerId: string, targetContainerId: string): boolean {
+    return (
+        !!runContainerId && !!targetContainerId && targetContainerId === runContainerId && user.hasInsertPermission()
```

Comment:
Is the `user.hasInsertPermission()` checking permissions in the target container?

Reply:
Yes, that is the intent. This isn't foolproof since someone could hand in an incorrect user instance (but that is true throughout the application at this time).